### ISSUE TITLE
Public role is too strong by default

### ISSUE THREAD

**Issue by xavier-GitHub76** [apache/superset#29709]
### Bug description

Hello,

to implement the 'public' role, the documentation (https://superset.apache.org/docs/security/#public) indicates that the PUBLIC_ROLE_LIKE variable must be updated and gives the example of using the GAMMA role.
It is also specified that the GAMMA role provides access for consultation.

By applying these recommendations and defining a “datasource access on” permission, an anonymous user can consult :
- a dashboard
- diagrams
- datasets
(by accessing the various main menus)

Visible elements are related to the “datasource access on” permission but the gamma permissions used like template are too strong.

An anonymous user can start creating a diagram and freely query a dataset via the diagram editing screen.
It can also export all diagram data.
However, it cannot save the diagram.

In order to achieve “read only” behavior, you should set up a role limited to the strict minimum and use it as a reference for the “public” role.

Here are the permissions I've identified as mandatory for this role:
- can dashboard permalink on Superset
- can read on Chart
- can read on Dashboard
- can dashboard on Superset
- can explore json on Superset
- can read on DashboardPermalinkRestApi
- can write on DashboardPermalinkRestApi
- can time range on Api
Of course, these permissions must be supplemented with the “datasource access on” permission.

With these permissions, an anonymous user will only be able to :
- access a dashboard via its permalink
- view the dashboard (and its diagrams)
- generate permalinks (on tabs, headers, dashboard and diagrams)
- filter a dashboard (including time ranges)
- export a dashboard (PDF or image)

Best regards

### How to reproduce the bug

1. Define a PUBLIC_ROLE_LIKE = GAMMA
2. On the "Public" role, add a “datasource access on” permission on a table
3. Visit superset like anonymous user
4. Click on "Dataset" menu
5. Click on public dataset
6. Edit a chart
7. Export data
8. Save chart (refused)

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [ ] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by rusackas** [comment]
Maybe it's time to add a new built-in public role (Delta?) for this purpose, but just not wire it up to anything by default? Curious also to hear if/how @dpgaspar would like to weigh in on the above.

**Comment by Arcelone** [comment]
I agree with @rusackas. I struggle since approximately 3 weeks to configure a proper "Public" role mixed or not with the RBAC feature. A proper minimal Public role should be included or at least documented. 

In addition I didn't found any documentation of permission. Is there one somewhere I have missed ? If not it's important to write one.

**Comment by Arcelone** [comment]
@xavier-GitHub76 what is the "can write to DashboardPermalinkRestApi" permission for? 
I tried without and I didn't found what was blocked.

**Comment by sfirke** [comment]
@Arcelone I'm guessing it's so that dashboard permalinks can be accessed.  E.g., accessing the dashboard at https://data.yourdomain.com/superset/dashboard/your-dashboard instead of https://data.yourdomain.com/superset/dashboard/101.  Did you try that?

**Comment by Arcelone** [comment]
I didn't get what you said. 

When you share the dashboard permalink it looks like : https://data.yourdomain.com/superset/dashboard/aiiduzgxykrofbzh 

The receiver paste it in his browser and here you go.

**Comment by sfirke** [comment]
Related discussion: https://github.com/apache/superset/discussions/25938

**Comment by xavier-GitHub76** [comment]
Without "can write to DashboardPermalinkRestApi"  : Issue #30004

**Comment by xavier-GitHub76** [comment]
The display of dashboard is impossible (redirection to login page) without one of  
- can dashboard on Superset
- can dashboard permalink on Superset
- can explore json on Superset
- can read on Chart
- can read on Dashboard
- can read on DashboardPermalinkRestApi

**Comment by xavier-GitHub76** [comment]
To download a chart as CSV/Excel/Image --> permission "can csv on Superset " (label not precise)

**Comment by rusackas** [comment]
Moving this to an Ideas discussion, since it sounds like that... a good idea, rather than a Superset bug. Holler if you want to open a PR and/or open a SIP to add a new role.
